src/lib/transaction_snark/test/account_timing/account_timing.ml

```diff
@@ -865,6 +865,57 @@ let%test_module "account timing check" =
               failwithf "Transaction failed: %s" (Error.to_string_hum err) ())
       |> Fn.flip Async.upon (fun () -> ())
 
+    let check_zkapp_failure expected_failure = function
+      | Ok
+          ( (parties_undo :
+              Mina_transaction_logic.Transaction_applied.Parties_applied.t)
+          , ( (local_state :
+                _ Mina_transaction_logic.Parties_logic.Local_state.t)
+            , _amount ) ) ->
+          (* we expect a Failed status, and the failure to appear in
+              the failure status table
+          *)
+          ( match With_status.status parties_undo.command with
+          | Applied _ ->
+              failwith "Expected transaction failure"
+          | Failed (failuress, _balances) ->
+              let failures = List.concat failuress in
+              if
+                not
+                  (List.equal Transaction_status.Failure.equal failures
+                     [ expected_failure ])
+              then
+                failwithf "Got unxpected transaction failure(s): %s"
+                  ( List.map failures ~f:Transaction_status.Failure.to_string
+                  |> String.concat ~sep:"," )
+                  () ) ;
+          let failure_statuses =
+            local_state.failure_status_tbl |> List.concat
```

Comment:
this should be equal to `failuress` and so the above check should cover the cases specified below. I'd include the expected failure in the error message

Reply:
adding an `assert` that those lists are equal